api/extension-guides/language-model-tutorial.md

```diff
@@ -9,13 +9,324 @@ MetaDescription: Tutorial that walks you through creating a VS Code extension th
+`;
+```
+
+This is a special prompt that instructs the language model on how to generate annotations. It also includes examples for how the model should format its response. This is what allows to to know what format the response will be in, so that we can parse it and display it as annotations. Now we can pass the prompt and the code with line numbers to the model.
+
+```ts
+const disposable = vscode.commands.registerTextEditorCommand('code-tutor.annotate', async (textEditor: vscode.TextEditor) => {
+
+  // Get the code with line numbers from the current editor
+  const codeWithLineNumbers = getCodeWithLineNumbers(textEditor);
+
+  // select the 4o chat model
+  let [model] = await vscode.lm.selectChatModels({
+    vendor: 'copilot',
+    family: 'gpt-4o',
+  });
+
+  // init the chat message
+  const messages = [
+    vscode.LanguageModelChatMessage.User(ANNOTATION_PROMPT),
+    vscode.LanguageModelChatMessage.User(codeWithLineNumbers),
+  ];
+});
+```
+
+We're ready to send the messages to the model and parse the response. First, we'll check to make sure that our model is ready to use. This accounts for when the extension is still loading or the user needs to grant permission or the extension to GitHub Copilot.
+
+```ts
+const disposable = vscode.commands.registerTextEditorCommand('code-tutor.annotate', async (textEditor: vscode.TextEditor) => {
+
+  // Get the code with line numbers from the current editor
+  const codeWithLineNumbers = getCodeWithLineNumbers(textEditor);
+
+  // select the 4o chat model
+  let [model] = await vscode.lm.selectChatModels({
+    vendor: 'copilot',
+    family: 'gpt-4o',
+  });
+
+  // init the chat message
+  const messages = [
+    vscode.LanguageModelChatMessage.User(ANNOTATION_PROMPT),
+    vscode.LanguageModelChatMessage.User(codeWithLineNumbers),
+  ];
+
+  // make sure the model is ready
+  if (model) {
+
+    // send the messages array to the model and get the response
+    let chatResponse = await model.sendRequest(messages, {}, new vscode.CancellationTokenSource().token);
+
+  }
+});
+```
+
+Then we'll send the messages to the model and get the response. Chat responses come in as fragments. These fragments usually contain single words, but sometimes they contain just punctuation. In order to display annotations as the response streams in, we want to wait until we have a complete annotation before we display it. Because of the way we have instructed our model to return its response, we know that when we see a closing `}` we have a complete annotation. We can then parse the annotation and display it in the editor.
+
+Add the following function above the `getCodeWithLineNumbers` method in the `extension.ts` file.
+
+```ts
+async function parseChatResponse(chatResponse: vscode.LanguageModelChatResponse, textEditor: vscode.TextEditor) {
```

Comment:
I would it funky that you first explain the parseChatResponse method, and only latter it gets into the flow. I would talk about this code at the end.